chainerx_cc/chainerx/native/native_device/hyperbolic.cc

```diff
@@ -0,0 +1,90 @@
+#include "chainerx/native/native_device.h"
+
+#include <cmath>
+#include <cstdint>
+
+#include "chainerx/array.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/native/elementwise.h"
+#include "chainerx/native/op_regist.h"
+#include "chainerx/numeric.h"
+#include "chainerx/routines/math.h"
+#include "chainerx/scalar.h"
+
+namespace chainerx {
+namespace native {
+namespace {
+
+template <typename T>
+struct NativeUnaryOp {
+    T (*func)(T);
+
+    NativeUnaryOp(T (*func)(T)) : func{func} {}
+
+    void operator()(int64_t /*i*/, T x, T& out) { out = func(x); }
+};
```

Comment:
The above part is slightly different from usual. Please review

Reply:
It looks not good to me because it seems to worsen performance because the inline expansion will not work well.

Reply:
Will revert it to the standard approach. Or should we make `struct` for `numeric ops` as done for Cuda